src/index.js

```diff
@@ -609,12 +610,40 @@ const members = {
       const getHtmlNodeOuterHtml = (node) => node.outerHTML;
       const getOuterHtmlByNodeType = {
         1: getHtmlNodeOuterHtml,
-        3: getEscapedTextContent,
         8: getCommentNodeOuterHtml,
       };
+      // Text nodes with these ancestors should be treated as raw text
+      // See section 8.4 of
+      // https://www.w3.org/TR/2008/WD-html5-20080610/serializing.html#html-fragment
+      const rawTextNodeAncestors = {
+        style: true,
+        script: true,
+        xmp: true,
+        iframe: true,
+        noembed: true,
+        noframes: true,
+        noscript: true,
+        plaintext: true,
+      };
 
       eachChildNode(this, node => {
-        const getOuterHtml = getOuterHtmlByNodeType[node.nodeType] || getHtmlNodeOuterHtml;
+        let getOuterHtml;
+        if (node.nodeType === 3) {
+          let parentNode = node.parentNode;
+          while (parentNode) {
+            if (parentNode.nodeType === 1 && // only for elements
+                parentNode.nodeName.toLowerCase() in rawTextNodeAncestors) {
+              getOuterHtml = getRawTextContent;
```

Comment:
I think it'd be good to be W3C spec compliant if it's not too much work or performance overhead. It looks like this would kill perf with large numbers of children because `children * levelsDeep`. Would it work the same if you instead detected the context based on the root node that `innerHTML` was called in instead of the children?